README.md

```diff
@@ -0,0 +1,94 @@
+# switchtec-dma
+
+Switchtec Gen4 PCIe switch introduces a new hardware DMA engine to offload
+expensive memory operations, such as large copies, from host processors. A
+complete solution of the Switchtec DMA involves not only the hardware DMA
+engine, but also embedded firmware and host software.
+
+The Switchtec DMA kernel module exports the access to the Switchtec DMA engine
+to the upper layer host software (DMA client).
+
+## Interface
+
+When configured, the Switchtec devices advertise a special DMA endpoint which
+enables the DMA functinality. BAR0 of the endpoint exports the firmware and
+hardware registers for the driver to configure and control the DMA engine.
+
+Switchtec dma driver maintains two queues in host memory space, a submission
+queue (SQ) and a completion queue (CQ), to communicate with the hardware DMA
+engine. Switchtec dma driver add Submission Elements to the SQ, on behalf of
+DMA client, when a DMA transfer is kicked off. DMA engine adds Completion
+Elements to the CQ when the DMA transfers is finished.
+
+## DMA mode
+
+Switchtec DMA driver can work in two modes, basic mode and fabric mode. With
+Switchtec PSX/PFX devices, the Switchtec DMA driver works in the basic mode.
```

Comment:
change "the basic mode" to "basic mode" (same with "the fabric mode" in the next sentence).